contracts/Everlong.sol

```diff
@@ -240,6 +240,44 @@ contract Everlong is IEverlong {
             );
     }
 
+    /// @notice Returns an approximate lower bound on the amount of assets
+    ///      received from redeeming the specified amount of shares.
+    /// @param _shares Amount of shares to redeem.
+    /// @return assets Amount of assets that will be received.
+    function previewRedeem(
+        uint256 _shares
+    ) public view override returns (uint256 assets) {
+        // Convert the share amount to assets.
+        assets = convertToAssets(_shares);
+
+        // If the amount of assets is less than Everlong's balance, no positions
+        // need to be closed and we can return.
+        uint256 balance = ERC20(_asset).balanceOf(address(this));
+        if (assets < balance) {
+            return assets;
+        }
+
+        // TODO: Evaluate whether we need mature position closing logic here.
+
+        // The amount of assets requires the closing of immature positions,
+        // during which losses can be incurred. Apply these losses to the
+        // amount of assets the redeemer will receive.
+        uint256 losses = _calcImmaturePositionLosses(assets - balance);
```

Comment:
Since this iterates over all positions (including mature positions), the "immature" in the name feels misleading. This also relates to the TODO comment above.

Reply:
Changed the naming and resolved the TODO